File: apps/portfolio-analytics/src/core/domain/entities/market_data.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass
class Quote:
    symbol: str
    price: float
    change: float
    change_rate: float
    volume: float | None = None
    timestamp: float = 0.0


@dataclass
class MarketData:
    data_id: UUID
    instrument_id: UUID
    timestamp: datetime
    open: float | None
    high: float | None
    low: float | None
    close: float
    volume: float | None
    change_pct: float | None

File: apps/server-go/internal/domain/market_data.go
package domain

import "time"

type MarketData struct {
	DataID      string    `json:"data_id"`
	InstrumentID string   `json:"instrument_id"`
	Timestamp   time.Time `json:"timestamp"`
	Open        *float64  `json:"open,omitempty"`
	High        *float64  `json:"high,omitempty"`
	Low         *float64  `json:"low,omitempty"`
	Close       float64   `json:"close"`
	Volume      *float64  `json:"volume,omitempty"`
	ChangePct   *float64  `json:"change_pct,omitempty"`
}
